packages/contracts-bedrock/contracts/EAS/resolver/SchemaResolver.sol

```diff
@@ -0,0 +1,182 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity 0.8.19;
+
+import { Semver } from "../../universal/Semver.sol";
+
+import { IEAS, Attestation } from "../IEAS.sol";
+import { InvalidEAS, uncheckedInc } from "../Common.sol";
+
+
+import { ISchemaResolver } from "./ISchemaResolver.sol";
+
+/**
+ * @title A base resolver contract
+ */
+abstract contract SchemaResolver is ISchemaResolver, Semver {
+    error AccessDenied();
+    error InsufficientValue();
+    error NotPayable();
+
+    // The global EAS contract.
+    IEAS internal immutable _eas;
```

Comment:
This can be a constant and the constant can be defined in `Predeploys.sol` so no need for a constructor argument

Reply:
The only downside is that it will make the interfaces much different than other deployments, while gas-wise, there's (almost) no difference. This can hinder projects that develop for other chains as well, so I'm not sure if it's a good fit here.

Reply:
Since this code lives in both the Optimism monorepo and your github repo, projects that are not OP Stack chains should use the code in your github repo. Using the hardcoded values is how we manage other predeploys and it will make deployment more simple